src/simplewallet/simplewallet.cpp

```diff
@@ -6442,16 +6438,16 @@ std::array<std::string, sizeof...(Prefixes)> eat_named_arguments(std::vector<std
 }
 
 // Parse a user-provided typestring value; if not provided, guess from the provided name and value.
-static std::optional<lns::mapping_type> guess_lns_type(tools::wallet2& wallet, std::string_view typestr, std::string_view name, std::string_view value)
+static std::optional<ons::mapping_type> guess_ons_type(tools::wallet2& wallet, std::string_view typestr, std::string_view name, std::string_view value)
 {
   if (typestr.empty())
   {
     if (tools::ends_with(name, ".loki") && (tools::ends_with(value, ".loki") || value.empty()))
-      return lns::mapping_type::lokinet;
-    if (!tools::ends_with(name, ".loki") && tools::starts_with(value, "05") && value.length() == 2*lns::SESSION_PUBLIC_KEY_BINARY_LENGTH)
-      return lns::mapping_type::session;
+      return ons::mapping_type::lokinet;
+    if (!tools::ends_with(name, ".loki") && tools::starts_with(value, "05") && value.length() == 2*ons::SESSION_PUBLIC_KEY_BINARY_LENGTH)
+      return ons::mapping_type::session;
```

Comment:
We should have a match after this that tries to parse it as a wallet address, and if it succeeds returns ons::mapping_type::wallet.  I think that would be enough to make the registration command work with a wallet without having to specify `type=wallet`.